services/frontend-v3/src/components/searchBar/SearchBar.jsx

```diff
@@ -0,0 +1,288 @@
+    try {
+      let results = await axios.get(
+        backendAddress + "api/option/getDevelopmentalGoals"
+      );
+
+      return results.data;
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  }
+
+  //Fetches options for branches select field in advanced search
+  async getBranch() {
+    try {
+      let results = await axios
+        .get(backendAddress + "api/option/getBranch")
+        .data.filter(elem => elem.description && elem.description.en);
+
+      return results;
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  }
+
+  //Fetches options for locations select field in advanced search
+  async getLocation() {
+    try {
+      let results = await axios.get(backendAddress + "api/option/getLocation");
+
+      return results.data;
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  }
+
+  //Fetches options for classifications select field in advanced search
+  async getClassification() {
+    try {
+      let results = await axios.get(
+        backendAddress + "api/option/getGroupLevel"
+      );
+
+      return results.data;
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  }
+
+  //Creates the basic inout field for basic search and puts its data into children array
+  getBasicField(data) {
+    const children = [];
+    const searchLabel = this.props.intl.formatMessage({
+      id: "button.search",
+      defaultMessage: "Search"
+    });
+    const { getFieldDecorator } = this.props.form;
+    children.push(
+      <Form.Item label={"Search"}>
+        {getFieldDecorator("Search", {})(<Input placeholder={searchLabel} />)}
+      </Form.Item>
+    );
+    return children;
+  }
+
+  //Creates the six fields for advanced search, along with their bilingual titles
+  getFields(data) {
+    const count = this.state.expand ? 6 : 0;
+
+    const { getFieldDecorator } = this.props.form;
+    const children = [];
+    var fieldCounter = 0;
+    const searchLabel = this.props.intl.formatMessage({
+      id: "button.search",
+      defaultMessage: "Search"
+    });
+    const labelArr = [
+      this.props.intl.formatMessage({
+        id: "advanced.search.form.name",
+        defaultMessage: "Name"
+      }),
+      this.props.intl.formatMessage({
+        id: "advanced.search.form.skills",
+        defaultMessage: "Skills"
+      }),
+      this.props.intl.formatMessage({
+        id: "advanced.search.form.branch",
+        defaultMessage: "Branch"
+      }),
+      this.props.intl.formatMessage({
+        id: "advanced.search.form.location",
+        defaultMessage: "Location"
+      }),
+      this.props.intl.formatMessage({
+        id: "advanced.search.form.classification",
+        defaultMessage: "Classification"
+      }),
+      this.props.intl.formatMessage({
+        id: "advanced.search.form.ex.feeder",
+        defaultMessage: "Ex Feeder"
+      })
+    ];
+    for (let i = 0; i < 10; i++) {
```

Comment:
I worry about doing this. In my opinion your getter should return data and that data should be passed to the view. In the view you can either code it in there or create another component that takes in the data as a prop. I'm worried about the maintainability of having "html" code in the controller.